src/tds/codec/column_data.rs

```diff
@@ -719,36 +720,59 @@ impl<'a> Encode<BytesMut> for ColumnData<'a> {
                 dst.extend_from_slice(&data);
             }
             ColumnData::String(Some(ref s)) if s.len() <= 4000 => {
-                dst.put_u8(VarLenType::NVarchar as u8);
-                dst.put_u16_le(8000);
-                dst.extend_from_slice(&[0u8; 5][..]);
+                // Allocate the headers and space for the UCS-2 bytes.
+                let mut c = Cursor::new(Vec::with_capacity(10 + s.bytes().len() * 2));
```

Comment:
I don't have a better suggestion, but `s.bytes().len() * 2` is likely to overallocate if your string is not mostly ascii.